include/ginkgo/core/matrix/dense.hpp

```diff
@@ -360,42 +362,29 @@ class Dense
     void row_permute(const Array<int64>* permutation_indices,
                      Dense* output) const;
 
-    /**
-     * Create a Dense matrix consisting of the given rows from this matrix.
-     *
-     * @param gather_indices  pointer to an array containing row indices
-     *                        from this matrix. It may contain duplicates.
-     * @return  Dense matrix on the same executor with the same number of
-     *          columns and `gather_indices->get_num_elems()` rows containing
-     *          the gathered rows from this matrix:
-     *          `output(i,j) = input(gather_indices(i), j)`
-     */
-    std::unique_ptr<Dense> row_gather(const Array<int32>* gather_indices) const;
+    std::unique_ptr<Dense> row_gather(
+        const Array<int32>* gather_indices) const override;
 
-    /**
-     * @copydoc row_gather(const Array<int32>*) const
-     */
-    std::unique_ptr<Dense> row_gather(const Array<int64>* gather_indices) const;
+    std::unique_ptr<Dense> row_gather(
+        const Array<int64>* gather_indices) const override;
 
-    /**
-     * Copies the given rows from this matrix into `row_gathered`
-     *
-     * @param gather_indices  pointer to an array containing row indices
-     *                        from this matrix. It may contain duplicates.
-     * @param row_gathered  pointer to a Dense matrix that will store the
-     *                      gathered rows:
-     *                      `output(i,j) = input(gather_indices(i), j)`
-     *                      It must have the same number of columns as this
-     *                      matrix and `gather_indices->get_num_elems()` rows.
-     */
```

Comment:
Unfortunately, this breaks interface. [See 1.4.0 dense.hpp](https://github.com/ginkgo-project/ginkgo/blob/v1.4.0/include/ginkgo/core/matrix/dense.hpp#L391)

Reply:
this interface is inherited from the EnableRowGather, which is also the reason why I need to create both for LinOp and Dense<VT> here

Reply:
I don't think it does, see the Line 377+ of this file. There are the `row_gather` functions that take a `Dense` input instead of a `LinOp`.

Reply:
Oh, yes, I missed that. A took the git diff at its face value. Thanks. Then, I don't think that should be an issue.